### ISSUE TITLE
Graph export feature

### ISSUE THREAD

**Issue by stapelberg** [prometheus/promdash#306]
It’d be neat if one could export a graph, in the same sense as taking a screenshot, but instead having it render (in high-quality) using the graphing library, with the data points coming from a static json file.

The use case is to embed these graphs in documentation, blog posts, etc.

**Comment by stuartnelson3** [comment]
have you looked at the embed functionality?

www.promdash.com/embed/{your-dashboard-name}

**Comment by juliusv** [comment]
I guess @stapelberg wants to have the data statically exported, with no queries happening to a Prometheus server when displaying the graph (e.g. when you do a blog post)?

**Comment by stuartnelson3** [comment]
hm, this could be an alternative to "link to graph", where the json blob is stored and an unmodifiable graph is rendered? interest @juliusv @grobie ?

**Comment by juliusv** [comment]
If I understand the request correctly, the graph should not even be served from PromDash at all anymore, but exported and made viewable without any PromDash around? Not sure... @stapelberg?

**Comment by stapelberg** [comment]
Correct: the graph should just be exported, with no assumptions about either prometheus or promdash still running. It should be entirely self-contained, suitable for embedding it in a blog post.

**Comment by matthiasr** [comment]
I second this.

Along similar lines, it would be great  to be able to export a statically compiled version of a particular dashboard view. Since we often treat Prometheus data as ephemeral, an embed or link with time is only useful for a limited time; I often want to preserve a particular view for eternity. This could be done by merging all relevant data into the HTML as javascript or, in the case of graphics (graphite graphs), [data URIs](http://en.wikipedia.org/wiki/Data_URI_scheme).

**Comment by grobie** [comment]
Sounds like we could need an embed icon for all these different types, opening some modal with the following options:

* standard link (js/css from current promdash instance, data from prometheus)
* public hands-free "gist" to be embedded everywhere (e.g. js/css from CDN, data from gist.github.com)
* raw html content to be copied and served by some server
* images anyone? :D http://html2canvas.hertzen.com/

**Comment by fiorix** [comment]
Any update on this?

**Comment by stuartnelson3** [comment]
i started spiking something out for this using http://fullscreen.github.io/squid/, but chances are pretty low this will happen unless the community does it

**Comment by fiorix** [comment]
I was trying to brute force using wkhtmltopdf but there's a lot of problems with that approach..

**Comment by fiorix** [comment]
Got the screenshot thing to work: https://gist.github.com/fiorix/dcceb49fb201fe5c218f

My use case is that both promdash and prometheus are behind a TLS connection with self-signed certificate, and require HTTP basic auth. The code in the gist above allows both, user/pass can be set in the URL.

**Comment by fiorix** [comment]
Ah, on the server I run it like this:

    xvfb-run -s "-screen 0 720x480x24" python website-screenshot --url=https://user:passwd@host/graph --sleep 5

Makes beautiful static images.
